app/models/concerns/with_status.rb

```diff
@@ -2,7 +2,7 @@ module WithStatus
   extend ActiveSupport::Concern
 
   included do
-    enum status: [:pending, :running, :passed, :failed]
+    enum status: [:pending, :running, :passed, :failed, :errored, :aborted, :passed_with_warnings]
```

Comment:
Can we turn this into something that support adding logic? There are many places where you are switching, perhaps it would be nicer to delegate that behaviour to the statuses.